apf/apfFieldData.cc

```diff
@@ -9,10 +12,13 @@ namespace apf {
 FieldData::~FieldData()
 {
 }
-
 FieldData* FieldData::clone()
 {
-  abort();
+  FieldData* newData = new TagDataOf<double>();
```

Comment:
Do we want to commit to all copied fields being created as mesh tags? 

Certainly all fields using the primary apf interfaces in apf.h are mesh tag fields but I'm not sure we want to have references to a subclass in a superclass. Should ```clone()``` be virtual and the type created depend on the subclass type?

Reply:
This is really up to you and @cwsmith. it was not pure virtual originally, so I just implemented it as that implied.